internal/slave/service.go

```diff
@@ -130,8 +158,11 @@ func (ss *slaveService) pollAndApplyChanges() {
 		case <-ss.replTckr.C:
 			ss.lg.Info("Current replication lag", zap.Uint64("ReplicationLag", ss.replLag))
 			ss.statsCli.Gauge("replication.lag", int64(ss.replLag))
-			if err := ss.applyChangesFromMaster(ss.maxNumChngs); err != nil {
-				ss.lg.Fatal("Unable to retrieve changes from master", zap.Error(err))
+			if err := ss.applyChangesFromMaster(ss.replConfig.MaxNumChngs); err != nil {
+				ss.lg.Error("Unable to retrieve changes from master", zap.Error(err))
+				if err := ss.replaceMasterIfInactive(); err != nil {
+					ss.lg.Error("Unable to replace master", zap.Error(err))
```

Comment:
Please add in some kind of backoff.

Reply:
if its due to bombarding of logs, set poll interval to 2 secs which fixes it

don't think functionally we should backoff here as master would be found as soon as service discovery cluster is available.

Reply:
During a network partition, even with a higher poll interval, it will keep failing continuously. IMO a higher poll interval would also lead to a lag in the slaves instead, we should have backoff in case of connection failures.

Reply:
yea i meant the higher poll interval in the ProcFile. ideally if slave is unable to replicate, slave is potentially useless. 

the repl poll interval is the guardrail set by the user for acceptable delays. I really don't think it would help if replication is stuck and we aren't retrying every poll interval